src/sundials_matrix_wrapper.cpp

```diff
@@ -399,50 +457,36 @@ void SUNMatrixWrapper::sparse_multiply(SUNMatrixWrapper &C,
      */
     
     sunindextype nnz = 0; // this keeps track of the nonzero index in C
-    auto Bx = B.data();
-    auto Bi = B.indexvals();
-    auto Bp = B.indexptrs();
-    
-    sunindextype bCol;
-    sunindextype bcolptr;
-    sunindextype ccolptr;
-    auto Cx = C.data();
-    auto Ci = C.indexvals();
-    auto Cp = C.indexptrs();
-    
-    sunindextype m = nrows;
-    sunindextype n = B.columns();
```

Comment:
That was on purpose, to avoid the call to `B.columns()` in every iteration, because afaik it wasn't as cheap as expected.